mycroft/skills/settings.py

```diff
@@ -195,18 +270,37 @@ def __setitem__(self, key, value):
             return super(SkillSettings, self).__setitem__(key, value)
 
     def _load_settings_meta(self):
-        """ Loads settings metadata from skills path. """
+        """ Load settings metadata from the skill folder.
+
+        If no settingsmeta exists a basic settingsmeta will be created
+        containing a basic identifier.
+
+        Returns:
+            (dict) settings meta
+        """
         if isfile(self._meta_path):
             try:
                 with open(self._meta_path, encoding='utf-8') as f:
                     data = json.load(f)
-                return data
             except Exception as e:
-                LOG.error("Failed to load setting file: "+self._meta_path)
+                LOG.error("Failed to load setting file: " + self._meta_path)
                 LOG.error(repr(e))
                 return None
         else:
-            return None
+            data = copy.copy(BLANK_META)
+
+        # Add Information extracted from the skills-meta.json entry for the
+        # skill.
+        skill_gid, title = build_global_id(self._directory, self.config)
+        data['skill_gid'] = skill_gid
+        data['title'] = (title or data.get('name') or
```

Comment:
@chrisveilleux did we agree to "title" or "display_name" on this?  This is the settingmeta.json

@forslund let's also remove leading/trailing skill from the camel_case_split.  E..g "PandoraMusicSkill" becomes "Pandora Music".  I think this is better 99% of the time, and they 1% can explicitly define a "name" to force "Skill" to be there if they really need it..

Reply:
I've updated to remove leading / trailing "Skill" and renamed the entry to `display_name`